File: GXpert/GXpert.Web/Modules/Syllabus/Class/ClassRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using Serenity.Extensions.Entities;
using System;
using System.ComponentModel;
using System.Reflection.Metadata;

namespace GXpert.Syllabus;

[ConnectionKey("Default"), Module("Syllabus"), TableName("Classes")]
[DisplayName("Class"), InstanceName("Class")]
[ReadPermission("Administration:General")]
[ModifyPermission("Administration:General")]
[ServiceLookupPermission("Administration:General")]
[LookupScript("Syllabus.Class")]
public sealed class ClassRow : LoggingRow<ClassRow.RowFields>, IIdRow, INameRow
{
    const string jCourse = nameof(jCourse);

    [DisplayName("Id"), Identity, IdProperty, LookupInclude]
    public int? Id { get => fields.Id[this]; set => fields.Id[this] = value; }

    [DisplayName("Title"), Size(500), NotNull, QuickSearch, NameProperty]
    public string Title { get => fields.Title[this]; set => fields.Title[this] = value; }

    [DisplayName("Description"), Size(2000)]
    public string Description { get => fields.Description[this]; set => fields.Description[this] = value; }

    [DisplayName("Sort Order"),NotNull]
    public short? SortOrder { get => fields.SortOrder[this]; set => fields.SortOrder[this] = value; }

    [DisplayName("Weightage")]
    public float? Weightage { get => fields.Weightage[this]; set => fields.Weightage[this] = value; }

    [DisplayName("Thumbnail"), Size(500),FileUploadEditor]
    public string Thumbnail { get => fields.Thumbnail[this]; set => fields.Thumbnail[this] = value; }

    [DisplayName("Course"), NotNull, ForeignKey("Course","Id"), LeftJoin(jCourse)]
    [ServiceLookupEditor(typeof(CourseRow)), LookupInclude]
    public int? CourseId { get => fields.CourseId[this]; set => fields.CourseId[this] = value; }

    [DisplayName("CourseName"), Expression($"{jCourse}.[Title]")]
    public string CourseName { get => fields.CourseName[this]; set => fields.CourseName[this] = value; }

    [DisplayName("Is Active"), DefaultValue(1)]
    public bool? IsActive { get => fields.IsActive[this]; set => fields.IsActive[this] = value; }

    public class RowFields : LoggingRowFields
    {
        public Int32Field Id;
        public StringField Title;
        public StringField Description;
        public Int16Field SortOrder;
        public SingleField Weightage;
        public StringField Thumbnail;
        public Int32Field CourseId;
        public StringField CourseName;
        public BooleanField IsActive;

    }
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Syllabus/ClassRow.ts
import { getLookup, getLookupAsync, fieldsProxy } from "@serenity-is/corelib";

export interface ClassRow {
    Id?: number;
    Title?: string;
    Description?: string;
    SortOrder?: number;
    Weightage?: number;
    Thumbnail?: string;
    CourseId?: number;
    CourseName?: string;
    IsActive?: boolean;
    InsertUserId?: number;
    InsertDate?: string;
    UpdateUserId?: number;
    UpdateDate?: string;
}

export abstract class ClassRow {
    static readonly idProperty = 'Id';
    static readonly nameProperty = 'Title';
    static readonly localTextPrefix = 'Syllabus.Class';
    static readonly lookupKey = 'Syllabus.Class';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<ClassRow>('Syllabus.Class') }
    static async getLookupAsync() { return getLookupAsync<ClassRow>('Syllabus.Class') }

    static readonly deletePermission = 'Administration:General';
    static readonly insertPermission = 'Administration:General';
    static readonly readPermission = 'Administration:General';
    static readonly updatePermission = 'Administration:General';

    static readonly Fields = fieldsProxy<ClassRow>();
}